app/rest/rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationRevisionHandler.java

```diff
@@ -0,0 +1,208 @@
+import javax.ws.rs.core.UriInfo;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class IntegrationRevisionHandler extends BaseHandler
+    implements Lister<IntegrationRevision>, Getter<IntegrationRevision>, Creator<IntegrationRevision>, Deleter<IntegrationRevision>, Updater<IntegrationRevision>, Validating<IntegrationRevision> {
+
+    private final EncryptionComponent encryptionSupport;
+
+    private final Validator validator;
+
+    public IntegrationRevisionHandler(final DataManager dataMgr, final Validator validator, final EncryptionComponent encryptionSupport) {
+        super(dataMgr);
+        this.validator = validator;
+        this.encryptionSupport = encryptionSupport;
+    }
+
+    @Override
+    public Kind resourceKind() {
+        return Kind.IntegrationRevision;
+    }
+
+
+    @Path("/{id}/{version}")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public IntegrationRevision get(@NotNull @PathParam("id") @ApiParam(required = true) String id, @NotNull @PathParam("version") @ApiParam(required = true) String version) {
+        String compositeId = id +"#" + version;
+        IntegrationRevision integrationRevision = Getter.super.get(compositeId);
+
+        if (integrationRevision.getCurrentState() == IntegrationRevisionState.Undeployed ||
+            integrationRevision.getTargetState() == IntegrationRevisionState.Undeployed) {
+            //Not sure if we need to do that for both current and desired status,
+            //but If we don't do include the desired state, IntegrationITCase is not going to pass anytime soon. Why?
+            //Cause that test, is using NoopHandlerProvider, so that means no controllers.
+            throw new EntityNotFoundException(String.format("Integration revision %s has been deleted", integrationRevision.getId()));
+        }
+
+        //fudging the timesUsed for now
+        if (integrationRevision.getCurrentState() == IntegrationRevisionState.Active) {
+            return new IntegrationRevision.Builder().createFrom(integrationRevision)
+                .timesUsed(BigInteger.valueOf(new Date().getTime()/1000000))
+                .build();
+        }
+
+        return integrationRevision;
+    }
+
+
+    @Override
+    public ListResult<IntegrationRevision> list(UriInfo uriInfo) {
+        return getDataManager().fetchAll(
+            IntegrationRevision.class,
+            new IntegrationRevisionHandler.DeletedFilter(),
+            new ReflectiveSorter<>(IntegrationRevision.class, new SortOptionsFromQueryParams(uriInfo)),
+            new PaginationFilter<>(new PaginationOptionsFromQueryParams(uriInfo))
+        );
+    }
+
+    @Override
+    public IntegrationRevision create(@Context SecurityContext sec, @ConvertGroup(from = Default.class, to = AllValidations.class) final IntegrationRevision integrationRevision) {
```

Comment:
Access to this API should be blocked.  Deployments get created as a result of publishing an integration.  There should be no REST api to this.